lib/wallet/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -771,6 +798,19 @@ class Show from => MkKeyFingerprint (key :: Depth -> Type -> Type) from where
 data ErrMkKeyFingerprint key from
     = ErrInvalidAddress from (Proxy key) deriving (Show, Eq)
 
+data AddressParts = AddressParts
+    { addrType :: Word8
+    , addrNetwork :: Word8
+    , rest :: ByteString
+    } deriving (Show,Eq)
+
+toAddressParts :: Address -> AddressParts
+toAddressParts (Address bytes) = AddressParts {..}
+  where
+    (fstByte, rest) = first BS.head $ BS.splitAt 1 bytes
+    addrType = fstByte .&. 0b11110000
+    addrNetwork = fstByte .&. 0b00001111
```

Comment:
`addrNetwork` may be wrong for Byron addresses? (was that the recent error?) https://github.com/input-output-hk/cardano-ledger/blob/6ef232e99ccd055180363673f814a21ae0ba1072/libs/cardano-ledger-core/src/Cardano/Ledger/Address.hs#L556 This seems like a hidden, untyped gotcha.

Reply:
we use it only for shelley and multisig style. byron and icarus stay as before. only in case of shelley and multisig it is used in isOurs and sequential state.

Reply:
If it's only supposed to work for Shelley addresses, I think it deserves to be mentioned in a doc comment

Reply:
fair point, done